packages/core/src/indexing/database-index-api.ts

```diff
@@ -201,17 +208,26 @@ export class DatabaseIndexApi<DateType = Date | number> {
   /**
    * Ensures that the given model StreamID can be queried and throws if not.
    */
-  assertModelQueryable(modelStreamId: StreamID | string): void {
-    // TODO(NET-1630) Throw if historical indexing is in progress
-    if (!this.allowQueriesBeforeHistoricalSync) {
-      throw new IndexQueryNotAvailableError(modelStreamId)
-    }
-
+  async assertModelQueryable(modelStreamId: StreamID | string): Promise<void> {
     const model = modelStreamId.toString()
-    if (this.modelsToIndex.find((indexedModel) => indexedModel.toString() == model) == undefined) {
+    const foundModelToIndex = this.modelsToIndex.find(
+      (indexedModel) => indexedModel.toString() == model
+    )
+    if (foundModelToIndex == undefined) {
       const err = new Error(`Query failed: Model ${model} is not indexed on this node`)
       this.logger.debug(err)
       throw err
+    } else {
+      await this.assertModelIndexable(modelStreamId)
+    }
+  }
+
+  async assertModelIndexable(modelStreamId: StreamID | string): Promise<void> {
```

Comment:
I don't really understand the division in names between `assertModelQueryable` and `assertModelIndexable`.  This function only runs on models that are already being actively indexed and just checks whether queries should be allowed through or not, so seems weird to me to call it assertModelsIndexable when the caller (assertModelsQueryable) has already asserted that the model is actually actively indexed

Reply:
`assertModelQueryable` no longer calls `assertModelIndexable`. That along the lines of what you were thinking?

Reply:
except it's still weird to call this `assertModelIndexable`.  It only runs on models that are already being indexed

Reply:
maybe just call it `assertNoOngoingSyncs` and put it back into being called by `assertModelsQueryable`?

Reply:
you could even take the current body of `assertModelsQueryable` and move it to a new helper called `assertModelsAreIndexed`.  Then the body of `assertModelsQueryable` would just be 2 lines long, calling the two helper functions for the two checks that determine if a model is queryable (that it is indexed and that there are no ongoing syncs)